src/common/m_precision_select.f90

```diff
@@ -0,0 +1,41 @@
+!> @file m_precision_select.f90
+!> @brief Contains module m_precision_select
+
+!> @brief This file contains the definition of floating point used in MFC
+module m_precision_select
+#ifdef MFC_MPI
+    use mpi                    !< Message Passing Interface (MPI) module
+#endif
+
+    implicit none
+
+    ! Define the available precision types
+    integer, parameter :: single_precision = selected_real_kind(6, 37)
+    integer, parameter :: double_precision = selected_real_kind(15, 307)
+
+    ! Set the working precision (wp) to single or double precision
+    integer, parameter :: wp = double_precision  ! Change this to single_precision if needed
+
+#ifdef MFC_MPI
+    ! Declare mpi_p as a module variable
+    integer :: mpi_p
+#else
+    integer, parameter :: mpi_p = -100
+#endif
+
+contains
+
+    ! Subroutine to initialize mpi_p based on wp
+    subroutine initialize_precision()
+#ifdef MFC_MPI
+        if (wp == single_precision) then
+            mpi_p = MPI_FLOAT
+        else if (wp == double_precision) then
+            mpi_p = MPI_DOUBLE_PRECISION
+        else
+            stop 'Unsupported precision kind.'
+        end if
+#endif
```

Comment:
I second @wilfonba's comments. @sbryngelson do we expect people to be able to run simulations in fp32 mode? Or are we expecting to only make subset of GPU kernels use single precision? If it's the former, it would be nice if we had a build toggle for this feature.

Reply:
@henryleberre I think for this PR, even getting single precision to work for all compute and storage (and being able to easily switch between build types) is the relevant thing. Eventually, I think we will want to do storage in fp32 (or even less) and compute in fp64 or 32 (perhaps even selected at the single kernel level). But it seems we need to do this step first. 

So, long story short, yes there should be a build flag for precision.